Automattic/wp-calypso
# Components: Remove SingleChildCSSTransitionGroup

This pull request seeks to remove the `<SingleChildCSSTransitionGroup />`, our custom fork of the `react-addons-css-transition-group` package component. It had originally been implemented to add support for (a) explicit transition timeouts and (b) a callback to be invoked after the child had been unmounted. The first of these was satisfied in React 0.14 by the [introduction of `transitionEnterTimeout` and `transitionLeaveeTimeout`](https://facebook.github.io/react/blog/2015/09/10/react-v0.14-rc1.html#new-deprecations-introduced-with-a-warning). The latter has been reimplemented here using the [component ref function callback](https://facebook.github.io/react/docs/more-about-refs.html) which is invoked with a `null` value when a component is unmounted. The `onClosed` prop is currently only used by the [Reader full post dialog](https://github.com/Automattic/wp-calypso/blob/cc130642303b102542e19693b9f70b7c04c41d07/client/reader/full-post/main.jsx#L371) in [unmounting the dialog](https://github.com/Automattic/wp-calypso/blob/cc130642303b102542e19693b9f70b7c04c41d07/client/reader/full-post/controller.js#L42-L44).

The changes here also resolve the following DOM prop warnings:

![image](https://cloud.githubusercontent.com/assets/1779930/17598674/9421451c-5fc9-11e6-917c-33f7bf6f242c.png)

__Testing instructions:__

Verify that there are no regressions in the display of a dialog, notably in the viewing and closing of a Reader full post dialog.

cc @blowery 

Test live: https://calypso.live/?branch=remove/single-child-css-transition-group
